gabrielgomesml/training_project_server
# Feature/movies-crud

## What I did

- Create Movies table;
- Create a table to keep users and movies relation;
- Create a table to keep genres and movies relation;
- Add query params to search movies by userId.

## How to test

- First of all make sure you have the docker installed on your machine
- Open the project and use the command `nvm use`
- Run a `yarn` or `yarn install`
- Run the command `docker-compose up -d` to create a docker container with the postgres database
- After that run the command `yarn typeorm migration:run` or `yarn migration` to update the database with migrations and `yarn start` for run the project
- Open insomnia, Postman or other similar softwares and create more than one Movie object with a POST in '/movies' route;
- Create a connection between a movie object that you created and an user with a POST in '/movies-users' route;
- User the route '/movies-users-user-id/${userId}' to find only the movies related to this user;
- Also, in the same route above, you can search between the movies related to this user by the movie title, you just need to add ?title=${text...}

Examples:

"/movies"

```
{
	"title": "Movie1",
	"synopsis": "This movie is about...",
	"release_year": "2020"
}
```

```
{
	"title": "Movie2",
	"synopsis": "This movie is about...",
	"release_year": "2021"
}
```

```
{
	"title": "Movie3",
	"synopsis": "This movie is about...",
	"release_year": "2022"
}
```

"/movies-users"

```
{
	"movie_id": [PUT HERE YOUR MOVIE1 ID],
	"user_id": [PUT HERE AN USER ID]
}
```

```
{
	"movie_id": [PUT HERE YOUR MOVIE2 ID],
	"user_id": [PUT HERE AN USER ID]
}
```

"movies-users-user-id/[PUT HERE THE USER ID]" - You should see only Movie1 and Movie2.

"movies-users-user-id/[PUT HERE THE USER ID]?text=Movie" - You shold see both, Movie1 and Movie2.
"movies-users-user-id/[PUT HERE THE USER ID]?text=Movie2" - You shold see only Movie2.
